inc/XamlApplication.h

```diff
@@ -105,4 +107,47 @@ namespace winrt {
   {
     return winrt::make<winrt::CppXaml::implementation::XamlApplication>(std::initializer_list<winrt::Windows::UI::Xaml::Markup::IXamlMetadataProvider>{providers...});
   }
+}
+
+#pragma comment(lib, "onecoreuap_apiset.lib")
+
+namespace cppxaml {
+  namespace details {
+    PWSTR g_winUIDepId{};
+    PACKAGEDEPENDENCY_CONTEXT g_ctx{};
+  }
+
+  /// <summary>
+  /// Dynamically initializes WinUI 2. This API works on Windows 11 and newer.
+  /// Adds the WinUI 2 framework package to the process's package graph.
+  /// This enables unpackaged applications to use the stable (non-prerelease) WinUI 2 NuGet package.
+  /// </summary>
+  /// <param name="minorVersion">Optional minor version. Defaults to 8 (for 2.8)</param>
+  void InitializeWinUI(uint8_t minorVersion = 8) {
+
+    if (details::g_winUIDepId == nullptr) {
+      PACKAGE_VERSION minVersion{};
+      minVersion.Major = minorVersion;
+
+      constexpr const std::wstring_view winuiPackageFamilyNameTemplate{ L"Microsoft.UI.Xaml.2.%d_8wekyb3d8bbwe" };
+      wchar_t pfn[PACKAGE_FAMILY_NAME_MAX_LENGTH]{};
+      winrt::check_hresult(StringCchPrintf(pfn, std::size(pfn), winuiPackageFamilyNameTemplate.data(), minorVersion));

```

Comment:
printf expects null terminated string, `wil::zwstring_view` gives you that.
or use `wil::str_printf<std::wstring>(...)` to form the family name. need to include wil/stl.h before wil/resource.h to use the std::wstring form of str_printf.

Reply:
Or move to C++20 and use `std::format` ;)

Reply:
I'll just use a string literal to get the null-terminated string template 😛 . Ideally everyone would move to cpp20 but... point-in-time